typing/typecore.ml

```diff
@@ -1358,7 +1358,7 @@ let rec type_pat ~constrs ~labels ~no_existentials ~mode ~explode ~env
                   pat_extra = extra :: p.pat_extra}
         in k p)
   | Ppat_type lid ->
-      let (path, p,ty) = build_or_pat !env loc lid.txt in
+      let (path, p,ty) = build_or_pat !env lid.loc lid.txt in
```

Comment:
One should probably pass both `loc`, the location of the pattern (for errors related to the pattern); and `lid` (including `lid.loc`) to be able to report errors related to the lookup of this identifier.